python/torch_mlir/csrc/base_lazy_backend/mlir_lowering_context.cpp

```diff
@@ -0,0 +1,255 @@
+  auto cu = std::make_shared<torch::jit::CompilationUnit>();
+  // IMPORTANT: We pass in a COPY of the graph into create_function, since it
+  //            may get mutated in the process.
+  auto jit_fn = cu->create_function(std::move(name), std::move(graph_->copy()));
+
+  // Generate MLIR.
+  MlirOperation func_op =
+      torch_mlir::importJitFunctionAsFuncOp(mlir_context_, jit_fn);
+
+  // TODO(henrytu): Inject tensor shapes into func_op
+  return std::make_shared<TorchMlirComputation>(func_op, mlir_context_, graph_);
+}
+
+torch::jit::Value* TorchMlirLoweringContext::GetOutputOp(const Output& output) {
+  PRINT_FUNCTION();
+
+  auto it = emitted_outputs_.find(output);
+  if (it == emitted_outputs_.end()) {
+    auto post_order = Util::ComputePostOrder(output.node, &emit_status_);
+    for (auto node : post_order) {
+      bool ok = lowering_->Lower(node);
+      TORCH_CHECK(ok, "Failed to lower: ", node->ToString());
+    }
+    // At this point the output better be present, otherwise there is an issue
+    // with the lowering code.
+    it = emitted_outputs_.find(output);
+    TORCH_CHECK(
+        it != emitted_outputs_.end(),
+        "No MLIR operation emitted for output: ", output.ToString());
+  }
+  return it->second;
+}
+
+void TorchMlirLoweringContext::AssignOutputOp(
+    const Output& output, torch::jit::Value* op) {
+  PRINT_FUNCTION();
+
+  auto torch_mlir_node =
+      NodeCast<TorchMlirNode>(output.node, output.node->op());
+  if (!torch_mlir_node->getPythonStacktrace().empty()) {
+    op->node()->s_(
+        c10::Symbol::attr("source"), torch_mlir_node->getPythonStacktrace());
+  }
+  emitted_outputs_[output] = std::move(op);
+}
+
+torch::jit::Value* TorchMlirLoweringContext::GetParameter(BackendDataPtr data) {
+  PRINT_FUNCTION();
+
+  if (!dynamic_cast<TorchMlirBackendData*>(data.get())) {
+    TORCH_CHECK(false, "Expected TorchMlirBackendData. Got some other BackendData type");
+  }
+  const auto mlir_data = std::static_pointer_cast<TorchMlirBackendData>(data);
+
+  BackendData::Handle handle = mlir_data->GetHandle();
+  auto it = parameters_map_.find(handle);
+
+  if (it == parameters_map_.end()) {
+    torch::jit::Value* param =
+        graph_->addInput(c10::str("p", parameters_.size()));
+
+    auto info = mlir_data->mlir_info();
+    if (info->scalar.has_value()) {
+      auto& scalar = info->scalar.value();
+      if (scalar.isFloatingPoint()) {
+        param->setType(c10::FloatType::get());
+      } else if (scalar.isIntegral(true)) {
+        param->setType(c10::IntType::get());
+      } else {
+        TORCH_CHECK(
+            false, "Unhandled scalar type: ", c10::toString(scalar.type()));
+      }
+    }
+
+    it = parameters_map_.emplace(handle, Parameter{param, parameters_.size()})
+             .first;
+    parameters_.push_back(mlir_data);
+  }
+
+  parameter_sequence_.push_back(it->second.index);
+  PRINT_DEBUG("Adding index " << it->second.index << " to parameter sequence");
```

Comment:
@henrytwo whoops. I think I missed removing this debug statement. Can you please remove it?